includes/widgets/accordion.php

```diff
@@ -78,6 +78,22 @@ protected function is_dynamic_content(): bool {
 		return false;
 	}
 
+	/**
+	 * Get style dependencies.
+	 *
+	 * Retrieve the list of style dependencies the widget requires.
+	 *
+	 * @since 3.24.0
+	 * @access public
+	 *
+	 * @return array Widget style dependencies.
+	 */
+	public function get_style_depends(): array {
```

Comment:
I would prefer if we could handle his logic in one of the widget base files.

Another question: could this lead to Core/Pro conflicts in any way?

Reply:
I agree, maybe in the base we can check if the CSS file exists, and if it does, load it in the `get_style_depends()`.

Maybe this way we also don't `need widgets_styles_names()`.

Reply:
@hein-obox  I'm trying to do the opposite. Not reduce the code, and hide everything in the widget base class.

I want the dependencies to be visible in the widget class, this way developers will see all the styles/scripts when looking at the widgets class. This would improve Developer Experience (DE), mainly for third-party developers.

As for Core/Pro conflicts, only if both widgets use the same name for style handles. But this is a general WordPress issue. The solutions is to use naming conventions.